hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java

```diff
@@ -16,35 +16,553 @@
+    }
+
+    private boolean shouldThrottleImplicitCreations(final int n, @NonNull final Instant now) {
+        final var manager = functionReqs.get(CRYPTO_CREATE);
+        return manager == null || !manager.allReqsMetAt(now, n, ONE_TO_ONE);
+    }
+
+    /*
+     * Rebuilds the throttle requirements based on the given throttle definitions.
+     *
+     * @param defs the throttle definitions to rebuild the throttle requirements based on
+     */
+    public void rebuildFor(@NonNull final ThrottleDefinitions defs) {
+        List<DeterministicThrottle> newActiveThrottles = new ArrayList<>();
+        EnumMap<HederaFunctionality, List<Pair<DeterministicThrottle, Integer>>> reqLists =
+                new EnumMap<>(HederaFunctionality.class);
+
+        for (var bucket : defs.throttleBuckets()) {
+            try {
+                final var utilThrottleBucket = new ThrottleBucket<>(
+                        bucket.burstPeriodMs(),
+                        bucket.name(),
+                        bucket.throttleGroups().stream()
+                                .map(this::hapiGroupFromPbj)
+                                .toList());
+                var mapping = utilThrottleBucket.asThrottleMapping(capacitySplitSource.getAsInt());
+                var throttle = mapping.getLeft();
+                var reqs = mapping.getRight();
+                for (var req : reqs) {
+                    reqLists.computeIfAbsent(req.getLeft(), ignore -> new ArrayList<>())
+                            .add(Pair.of(throttle, req.getRight()));
+                }
+                newActiveThrottles.add(throttle);
+            } catch (IllegalStateException badBucket) {
+                log.error("When constructing bucket '{}' from state: {}", bucket.name(), badBucket.getMessage());
+            }
+        }
+        EnumMap<HederaFunctionality, ThrottleReqsManager> newFunctionReqs = new EnumMap<>(HederaFunctionality.class);
+        reqLists.forEach((function, reqs) -> newFunctionReqs.put(function, new ThrottleReqsManager(reqs)));
+
+        functionReqs = newFunctionReqs;
+        activeThrottles = newActiveThrottles;
+
+        logResolvedDefinitions(capacitySplitSource.getAsInt());
+    }
+
+    /*
+     * Rebuilds the gas throttle based on the current configuration.
+     */
+    public void applyGasConfig() {
+        final var configuration = configProvider.getConfiguration();
+        final var contractsConfig = configuration.getConfigData(ContractsConfig.class);
+        if (contractsConfig.throttleThrottleByGas() && contractsConfig.maxGasPerSec() == 0) {
+            log.warn("Consensus gas throttling enabled, but limited to 0 gas/sec");
+            return;
+        }
+
+        final long capacity = contractsConfig.maxGasPerSec();
+        gasThrottle = new GasLimitDeterministicThrottle(capacity);
+
+        log.info(
+                "Resolved consensus gas throttle -\n {} gas/sec (throttling {})",
```

Comment:
⚠️ Hard-coding "consensus" here will result in some confusing logs, since this class also implements the frontend throttle. Imo we should make the type of throttle a constructor parameter.